clients/include/d_vector.hpp

```diff
@@ -76,13 +77,15 @@ class d_vector
             if(PAD > 0)
             {
                 // Copy guard to device memory before allocated memory
-                CHECK_HIP_ERROR(hipMemcpy(d, guard, sizeof(guard), hipMemcpyHostToDevice));
+                if(hipMemcpy(d, guard, sizeof(guard), hipMemcpyHostToDevice))
+                    std::cerr << "Error: hipMemcpy pre-guard copy failure." << std::endl;
```

Comment:
Does this ensure a failed test?

Reply:
With hipGetLastError added, now it does.